pyproject.toml

```diff
@@ -10,20 +10,21 @@ authors = [
     { name = "Shuntaro C. Aoki", email = "s_aoki@i.kyoto-u.ac.jp" }
 ]
 readme = "README.md"
-requires-python = ">= 3.6, < 3.12"
+requires-python = ">= 3.6, < 3.9"
 license = { file = "LICENSE" }
 keywords = ["neuroscience", "neuroimaging", "brain decoding", "fmri", "machine learning"]
 
 dependencies = [
-    "numpy>=1.20",
+    "numpy>=1.19",
     "scipy",
     "scikit-learn",
-    "h5py",
+    "h5py>3.0; python_version >= '3.7'",
+    "h5py<3.0; python_version < '3.7'",
     "hdf5storage",
     "pyyaml",
     "pandas",
     "tqdm",
-    "typing-extensions>=4.5",
+    "typing-extensions>=4.1.0",
```

Comment:
typing-extensions >= 4.5 was likely specified for a reason, so if this requirement is for legacy environments, please create branching logic for legacy cases.